File: Lab5/Sharp/Lab5/Lab5/IntArray.cs
using System;
using System.Linq;

namespace Lab5
{
    internal class IntArray : TArray<int>
    {
        public IntArray(int length)
        {
            if (length < 0) throw new ArgumentOutOfRangeException("length");

            _elems = new int[length];
        }
        public override void IncreaseAll(int value)
        {
            for (int i = 0; i < Count; i++)
            {
                _elems[i] += value;
            }
        }
        public override void DecreaseAll(int value)
        {
            for (int i = 0; i < Count; i++)
            {
                _elems[i] -= value;
            }
        }
        public override double GetAverage()
        {
            int sum = _elems.Sum();
            return (double)sum / _elems.Count();
        }

        public override void FillRandomly()
        {
            for (int i = 0; i < Count; i++)
            {
                _elems[i] = _random.Next(-50, 51);
            }
        }
    }
}

File: Lab5/Python/int_array.py
from random import *

from t_array import TArray


class IntArray(TArray):
    def get_average(self):
        sum_of_all = 0
        for i in self.items:
            sum_of_all += i
        return round(sum_of_all / self.size, 3)

    def __str__(self):
        return ' '.join([str(x).rjust(3, ' ') for x in self.items])

    def fill_randomly(self):
        for i in range(self.size):
            self.items[i] = randint(-25, 25)
